tasks/scripting/src/main/java/org/infinispan/scripting/impl/ScriptMetadataParser.java

```diff
@@ -79,11 +84,25 @@ public static ScriptMetadata parse(String name, String script) {
    private static Set<String> unarray(String s) {
       if (s.charAt(0) == '[') {
          String[] ps = s.substring(1, s.length() - 1).split("\\s*,\\s*");
-         Set<String> parameters = new HashSet<>();
+         Set<String> parameters = new HashSet<>(ps.length);
          Collections.addAll(parameters, ps);
          return parameters;
       } else {
          throw log.parametersNotArray();
       }
    }
+
+   private static Map<String, String> unobject(String s) {
+      if (s.charAt(0) == '{') {
+         String[] ps = s.substring(1, s.length() - 1).split("\\s*,\\s*");
+         Map<String, String> properties = new HashMap<>(ps.length);
+         for(String p : ps) {
+            String[] kv = p.split(":");
+            properties.put(kv[0].trim(), kv[1].trim());
```

Comment:
Since it uses an object notation, the key and value would be wrapped in quotes (`\"key\"` instead of `key`), is that a problem?

Reply:
"unobject" is a bit of a misnomer: we are using unquoted names and values, both here and in the array.